src/dune_rules/ocaml_flags.ml

```diff
@@ -64,11 +64,35 @@ let default_flags ~dune_version ~profile =
 
 type 'a t' =
   { common : 'a
+  ; keywords : string option * string list
```

Comment:
I see you mentioned not over-abstracting as per our hacking doc, but in this case I think a `Keywords.t` or similar is necessary so that the version and extra keywords can go in named fields. Otherwise I think it will be hard for future readers of this code to understand the meaning of the tuple elements.